JSTutorial/hw_7/task_2.js

```diff
@@ -0,0 +1,299 @@
+    {
+        id: 5,
+        name: "Предприятие 2",
+        departments: [
+            {
+                id: 6,
+                name: "Отдел разработки",
+                employees_count: 50,
+            },
+            {
+                id: 7,
+                name: "Отдел маркетинга",
+                employees_count: 20,
+            },
+            {
+                id: 8,
+                name: "Отдел охраны труда",
+                employees_count: 5,
+            },
+        ]
+    },
+    {
+        id: 9,
+        name: "Предприятие 3",
+        departments: [
+            {
+                id: 10,
+                name: "Отдел аналитики",
+                employees_count: 0,
+            },
+        ]
+    }
+]
+
+//1
+function countEmployees(departments) {
+    return departments.reduce(function(total, department) {
+        return total + department.employees_count;
+    }, 0);
+}
+console.log(countEmployees(enterprises[0].departments));
+const getCommonInfo = (company) => {
+    company.forEach(enterprise => {
+            console.log(`${enterprise.name} (${countEmployees(enterprise.departments)} сотрудников)`);
+            enterprise.departments.forEach(department => console.log(`- ${department.name} (${department.employees_count})`));
+    })
+}
+getCommonInfo(enterprises);
+
+//2
+function getFactoryByNameORId(key) {
+    let name = null;
+    enterprises.forEach(enterprise => {
+        enterprise.departments.forEach(department => {
+            if (department.id === key || department.name === key) name = enterprise.name;
+        })
+    })
+    return name;
+}
+console.log(getFactoryByNameORId("Отдел тестирования"))
+console.log(getFactoryByNameORId(6))
+
+//3
+function addEnterprise(name) {
+    const nejObj = {
+        name,
+        departments: [],
+        id : enterprises.length - 1,
+    }
+    enterprises.push(nejObj);
+}
+
+addEnterprise("Предприятие 4")
+console.log(enterprises);
+
+//4
+function getEnterpriseByIdOrName(key) {
+    let enterpriseObj = null;
+    enterprises.forEach(enterprise => {
+        if (enterprise.id === key || enterprise.name === key) enterpriseObj = enterprise
+    })
+    return enterpriseObj;
+}
+function addDepartment(id, name) {
+    const enterprise = getEnterpriseByIdOrName(id);
+    const nejObj = {
+        name,
+        employees_count: [],
+        id : enterprise.id + 12,
+    }
+    enterprise.departments.push(nejObj);
+}
+addDepartment(2, "Новый отдел");
+console.log(enterprises[3])
+
+//5
+function editDepartment(id, name) {
+    const enterprise = getEnterpriseByIdOrName(id);
+    enterprise.name = name;
+}
+editDepartment(2, "Измененное предприятие")
+console.log(enterprises[3])
+
+//6
+function deleteEnterprise(id) {
+    const index = enterprises.findIndex(enterprise => enterprise.id === id);
+    if (index !== -1) {
+        enterprises.splice(index, 1);
+    }
+}
+
+deleteEnterprise(2)
+console.log(enterprises)
+
+//7
+function getDepartmentIndexById(id) {
```

Comment:
Try specifying a search array as a function parameter so that it works only with the data that was passed, making it independent
```suggestion
function getDepartmentIndexById(id, arr) {
```